src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,137 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int START_CAPACITY = 10;
+    private static final int START_SIZE = 0;
+    private static final int NONE_ELEMENT_FIND = -1;
+    private Object[] array;
+    private int realSize;
+    private int capacity;
 
+    enum ActionMethod {
+        INSERT,
+        GET
     }
 
-    @Override
-    public void add(T value, int index) {
+    public ArrayList() {
+        array = new Object[START_CAPACITY];
+        realSize = START_SIZE;
+        capacity = START_CAPACITY;
+    }
+
+    private void isWrongIndex(int index, ActionMethod action)
+            throws ArrayListIndexOutOfBoundsException {
+        if (action == ActionMethod.GET) {
+            if (index < 0 || index >= realSize) {
+                throw new ArrayListIndexOutOfBoundsException("Index is out of range");
+            }
+        }
+        if (action == ActionMethod.INSERT) {
+            if (index < 0 || index > realSize) {
+                throw new ArrayListIndexOutOfBoundsException("Index is out of range");
+            }
+        }
```

Comment:
This check is needed only once in your code. Add index check to the method where you need it.